crates/moveos/src/vm/tx_argument_resolver.rs

```diff
@@ -0,0 +1,199 @@
+
+        if has_tx_context {
+            args.insert(0, self.to_vec());
+        }
+
+        Ok(args)
+    }
+}
+
+// TODO move to move_table_extension
+impl TxArgumentResolver for move_table_extension::NativeTableContext<'_> {
+    fn resolve_argument<S>(
+        &self,
+        _session: &SessionExt<S>,
+        _func: &LoadedFunctionInstantiation,
+        args: Vec<Vec<u8>>,
+    ) -> Result<Vec<Vec<u8>>, PartialVMError>
+    where
+        S: MoveResolverExt,
+    {
+        //TableContext do nothing to the arguments now.
+        Ok(args)
+    }
+}
+
+//TODO move to NativeObjectContext
+impl TxArgumentResolver for NativeObjectContext<'_> {
+    fn resolve_argument<S>(
+        &self,
+        session: &SessionExt<S>,
+        func: &LoadedFunctionInstantiation,
+        args: Vec<Vec<u8>>,
+    ) -> Result<Vec<Vec<u8>>, PartialVMError>
+    where
+        S: MoveResolverExt,
+    {
+        if args.len() != func.parameters.len() {
+            return Err(PartialVMError::new(
+                StatusCode::NUMBER_OF_ARGUMENTS_MISMATCH,
+            ));
+        }
+
+        let mut resolved_args = vec![];
+        for (i, (t, arg)) in func.parameters.iter().zip(args.iter()).enumerate() {
+            let is_object = is_object(session, t);
+            let arg = match is_object {
+                Some(t) => {
+                    //skip tx context and signer
+                    if is_tx_context(&t) {
+                        arg.clone()
+                    } else {
+                        let object_id = ObjectID::from_bytes(arg).map_err(|_e| {
+                            PartialVMError::new(StatusCode::NUMBER_OF_TYPE_ARGUMENTS_MISMATCH)
+                        })?;
+                        let object: Object = self.get_object(object_id)?.ok_or_else(|| {
+                            PartialVMError::new(StatusCode::NUMBER_OF_TYPE_ARGUMENTS_MISMATCH)
+                                .with_message(format!("Can not find object: {:?}", object_id))
+                        })?;
+                        match object.data {
+                            ObjectData::MoveObject(m) => m.contents,
+                            ObjectData::TableObject(_) => {
+                                return Err(PartialVMError::new(
+                                    StatusCode::NUMBER_OF_TYPE_ARGUMENTS_MISMATCH,
+                                )
+                                .with_message(format!(
+                            "Table object is not supported as argument currently, argument pos: {}",
+                            i + 1
+                        )))
+                            }
+                        }
+                    }
+                }
+                None => arg.clone(),
+            };
+            resolved_args.push(arg);
+        }
+
+        Ok(args)
+    }
+}
+
+fn is_signer(t: &Type) -> bool {
+    matches!(t, Type::Signer)
+}
+
+fn is_object<T>(session: &SessionExt<T>, t: &Type) -> Option<Arc<StructType>>
+where
```

Comment:
is_object -> get_object ?
I feel a function or a variable starts with `is` will be return `bool` type

Reply:
`None` -> false
`Some` -> true and return another result.